x-pack/plugins/observability_solution/infra/public/alerting/metric_threshold/rule_data_formatters.ts

```diff
@@ -5,14 +5,41 @@
  * 2.0.
  */
 
-import { ALERT_REASON } from '@kbn/rule-data-utils';
+import { ALERT_REASON, ALERT_RULE_PARAMETERS } from '@kbn/rule-data-utils';
 import { ObservabilityRuleTypeFormatter } from '@kbn/observability-plugin/public';
+import { LocatorPublic } from '@kbn/share-plugin/common';
+import type { AssetDetailsLocatorParams } from '@kbn/observability-shared-plugin/common';
+import { InventoryItemType, findInventoryModel } from '@kbn/metrics-data-access-plugin/common';
+import { SupportedAssetTypes } from '../../../common/asset_details/types';
 import { getMetricsViewInAppUrl } from '../../../common/alerting/metrics/alert_link';
-export const formatReason: ObservabilityRuleTypeFormatter = ({ fields }) => {
-  const reason = fields[ALERT_REASON] ?? '-';
 
-  return {
-    reason,
-    link: getMetricsViewInAppUrl(fields),
+export const getFormatReason = ({
```

Comment:
Can we please have a unit test for this logic?

Reply:
This needs tests indeed. will do.